src/serializer/modules.js

```diff
@@ -457,7 +458,9 @@ export class Modules {
       // Check for escaping property assignments, if none escape, we got an existing object
       let escapes = false;
       for (let [binding] of properties) {
-        if (!createdObjects.has(binding.object)) escapes = true;
+        let object = binding.object;
+        invariant(object instanceof ObjectValue);
```

Comment:
Why is this invariant needed now when it wasn't before?

Reply:
`binding.object` can be an AbstractValue, I added a return type of `Effects` to evaluateForEffects so flow may not have considered that possibility before.